src/transformState.js

```diff
@@ -5,7 +5,23 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const obj of actions) {
+    if (obj.type === 'addProperties') {
+      Object.assign(state, obj.extraData);
+    }
+
+    if (obj.type === 'removeProperties') {
+      for (const del of obj.keysToRemove) {
+        delete state[del];
+      }
+    }
+
+    if (obj.type === 'clear') {
+      for (const el in state) {
+        delete state[el];
```

Comment:
It is better to use [switch...case](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch) to go through all possible types